src/Core/src/Config/Proxy.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Spiral\Core\Config;
+
+class Proxy extends Binding
+{
+    /**
+     * @param class-string $interface
+     */
+    public function __construct(
+        protected readonly string $interface,
+        public readonly bool $singleton = false,
+    ) {
+    }
```

Comment:
I think it will be nice to check that `$interface` is an interface and thow InvalidArgumentException if not.

Reply:
Added